rust/main/agents/relayer/src/msg/metadata/routing.rs

```diff
@@ -30,10 +31,57 @@ impl MetadataBuilder for RoutingIsmMetadataBuilder {
             .build_routing_ism(ism_address)
             .await
             .map_err(|err| MetadataBuildError::FailedToBuild(err.to_string()))?;
-        let module = ism
-            .route(message)
+
+        let message_domain = KnownHyperlaneDomain::try_from(message.origin)
+            .map(|domain| domain.as_str().to_string())
+            // if its an unknown domain, use the raw u32 as a string
+            .unwrap_or_else(|_| message.origin.to_string());
+        let fn_key = "route";
+
+        // Depending on the cache policy, make use of the message ID
+        let params_cache_key = match self
+            .base_builder()
+            .ism_cache_policy_classifier()
+            .get_cache_policy(self.root_ism, ism.domain(), ModuleType::Routing)
             .await
-            .map_err(|err| MetadataBuildError::FailedToBuild(err.to_string()))?;
+        {
+            // To have the cache key be more succinct, we use the message id
+            IsmCachePolicy::IsmSpecific => (ism.address(), H256::zero()),
+            IsmCachePolicy::MessageSpecific => (ism.address(), message.id()),
+        };
+
+        let cache_result: Option<H256> = self
+            .base_builder()
+            .cache()
+            .get_cached_call_result(message_domain.as_str(), fn_key, &params_cache_key)
```

Comment:
the first param here should be the `ism.domain().name()`. this is meant to be the domain of the ISM that we are caching